tests/net/dhcpv4/src/main.c

```diff
@@ -401,59 +287,53 @@ struct net_pkt *prepare_dhcp_ack(struct net_if *iface, u32_t xid)
 
 static int parse_dhcp_message(struct net_pkt *pkt, struct dhcp_msg *msg)
 {
-	struct net_buf *frag = pkt->frags;
-	u8_t type;
-	u16_t offset;
+	net_pkt_cursor_init(pkt);
```

Comment:
I guess this is superfluous, net_send_data() and, if required, l2 will call that already.